### ISSUE TITLE
 3.08 on MacOS fluctuating performance.

### ISSUE THREAD

**Issue by mquantin** [GPUOpen-LibrariesAndSDKs/RadeonProRenderBlenderAddon#232]
AMD radeonProRender is very slow, unusable in viewport, slow and bad quality in rendering (looks like a badly printed image). Moreover it doesn' take into accound alphaBlend (transparency) of materials and renders everything shiny...

Is there something I did wrong? 

macOS 11.1 (20C69)
Model Name: iMac Pro (2017)
Processor Name: 3.2 GHz Intel Xeon W 8 cœurs
Memory: 64 Go 2666 MHz DDR4
Chipset Model: Radeon Pro Vega 64X 16 Go

**Comment by ardraeiss** [comment]
Hello.
Could you share the simple example scene you have issues with and how it looks like for you?

**Comment by mquantin** [comment]
Hello, thanks for your reply. 

[proRender.zip](https://github.com/GPUOpen-LibrariesAndSDKs/RadeonProRenderBlenderAddon/files/6203473/proRender.zip)

In that file you'll find:
- simple blend scene
- a cycles render on CPU : 39seconds
- an evee render on CPU : instantaneous
- a first ProRender on CPU+GPU : 19 seconds
- a second one. Same config. Just another render. Different aspect and much much slower process: 91 seconds
- a screenshot of the "render preview" mode with ProRender. Very laggy

So I discover the rendering time with proRender is also very fluctuating.

Please note also there is nothing mentionned about GPU in ">preferences > System". Is that normal? (i'm new with Apple devices)

Thank you for your help.

**Comment by bsavery** [comment]
Hi Matthieu.  We'll take a look at the issue, but in the future I think the title "Not functional at all" is a bit Hyperbolic and not very descriptive of your issue of fluctuating performance ;) I'll edit the title to be more descriptive.

Thanks!

**Comment by bsavery** [comment]
Also I forgot to mention can you try with the latest builds on https://github.com/GPUOpen-LibrariesAndSDKs/RadeonProRenderBlenderAddon/releases which is later than 3.0.8

**Comment by mquantin** [comment]
OK thanks for the edits. Still the issue here is:
"on MacOS 11.1 ProVega 64X 16 Go: very slow and bad quality for viewport and render images, unusable"
what about that title? 

As the number of pixel on my screen is huge (5K). i believe the viewport is kind of consuming GPU to render all these pixels. 
1. **Fluctuating rendering performances** : with viewport in _solid mode_, the render speed of an "image through camera" is stable, faster than cycle on CPU but still bad quality. This is the situation of the "first ProRender on CPU+GPU" I previously described.
2. **Viewport render preview mode**: Reducing the size of blender viewport window speeds up the viewport render, that remain  bad quality, much much slower that _cycles_ on CPU, unusable.

So the fuctuating performances of the " image through camera" renders seems to be a consequence of a very slow rendering on the viewport with _render preview mode_.

The 3.1.0 release give the same results unfortunatly. I catched an error in viewport _render preview mode_ (screenshot below) I never saw with 3.08 ProRender version.  Sorry I can't find logs with macOS...
<img width="800" alt="Capture d’écran 2021-03-29 à 10 37 37" src="https://user-images.githubusercontent.com/11961831/112814622-c547de80-907f-11eb-9489-870baa5f875f.png">


**Performance comparison**: with blender on Windows and a cheap GPU, with _cycles_, viewport _render preview mode_ is super fluid and "image through camera" renders are fast and good...
What kind of render result / viewport screen do you get with the exemple scene i provided?

**Comment by bsavery** [comment]
Hi Matthieu

> OK thanks for the edits. Still the issue here is:
> "on MacOS 11.1 ProVega 64X 16 Go: very slow and bad quality for viewport and render images, unusable"
> what about that title?
> 
> As the number of pixel on my screen is huge (5K). i believe the viewport is kind of consuming GPU to render all these pixels.
> 

We downscale the viewport resolution for speed, you can disable this in the viewport settings

> 1. **Fluctuating rendering performances** : with viewport in _solid mode_, the render speed of an "image through camera" is stable, faster than cycle on CPU but still bad quality. This is the situation of the "first ProRender on CPU+GPU" I previously described.
I would use GPU only, not CPU + GPU for viewport rendering.  

> 2. **Viewport render preview mode**: Reducing the size of blender viewport window speeds up the viewport render, that remain  bad quality, much much slower that _cycles_ on CPU, unusable.

> 
> So the fuctuating performances of the " image through camera" renders seems to be a consequence of a very slow rendering on the viewport with _render preview mode_.

So are you rendering two views at once here?  I'm not clear.  Or are you saying that one camera view vs the other is slower?
> 
> The 3.1.0 release give the same results unfortunatly. I catched an error in viewport _render preview mode_ (screenshot below) I never saw with 3.08 ProRender version. Sorry I can't find logs with macOS...
> <img alt="Capture d’écran 2021-03-29 à 10 37 37" width="800" src="https://user-images.githubusercontent.com/11961831/112814622-c547de80-907f-11eb-9489-870baa5f875f.png">
> 

Disable the viewport denoising and upscaling in the viewport settings (see #241) 
> **Performance comparison**: with blender on Windows and a cheap GPU, with _cycles_, viewport _render preview mode_ is super fluid and "image through camera" renders are fast and good...
> What kind of render result / viewport screen do you get with the exemple scene i provided?

**Comment by gwhobbs** [comment]
While ProRender GPU seems to be much faster on final renders than Cycles CPU, I am also observing the reverse for interactive viewport renders. One thing that seems to help significantly when using high-resolution displays is reducing the *max* resolution scale. I added a custom control for this and modified the add-on to support it locally, and am getting much better interactive performance (still lags quite a bit making changes when compared to Cycles CPU, but much more usable). 

![Max Resolution Scale](https://user-images.githubusercontent.com/4071958/134821841-5ef2a932-5446-4605-86b7-4665b0ddea90.png)

I suspect (but have not verified) that Cycles may be checking somehow whether the OS is using 2x UI scaling—and if so, rendering the preview at 50%. If true, this would at least explain some of the performance difference.

**Comment by bsavery** [comment]
"I suspect (but have not verified) that Cycles may be checking somehow whether the OS is using 2x UI scaling—and if so, rendering the preview at 50%. If true, this would at least explain some of the performance difference."

Huh.  That's an interesting theory.  I'm not sure how we could detect and do anything from python though...

**Comment by gwhobbs** [comment]
Did a quick follow-up test with Cycles Viewport Pixel Size, which is by default set to `Automatic`. A close inspection of the viewport looks like it's a ~2x scaled image. Changing it from `Automatic` to `2x` does not trigger a viewport update, but changing it from `Automatic` to anything else does. I haven't yet verified, though, that this does not happen in other configurations, such as when 2x system UI scaling is disabled. 

However, I also found `bpy.context.preferences.system.ui_scale` via [this thread](https://devtalk.blender.org/t/proper-way-to-draw-viewport-text-when-using-hi-dpi-displays/6804/2), which is showing as `2.0` for me. I have a hunch that this is what they’re using. 

Default behavior aside, the ability to override the maximum viewport resolution is pretty useful, and is currently possible in e.g. Cycles and RenderMan for Blender. What are your thoughts on possibly including it in some form in a future release? Maybe (as opposed to the way I tried it initially) it would be best as a dropdown with 1x, 2x, 4x, 8x, similar to Cycles—maybe with an Automatic option—and be presented outside the "Adapt Viewport Resolution" settings group?

Cycles viewport scaling preferences:
![Screen Shot 2021-10-05 at 10 25 47 PM](https://user-images.githubusercontent.com/4071958/136135949-7854ea79-a5fb-4a6c-aa26-e74ef018caa5.png)

RenderMan for Blender:
![image](https://user-images.githubusercontent.com/4071958/136138319-5c2ee8ab-aae8-4405-b876-40cd6ff25263.png)
